src/main/java/org/betonquest/betonquest/events/FolderEvent.java

```diff
@@ -73,9 +85,14 @@ protected Void execute(final Profile profile) throws QuestRuntimeException {
                 BetonQuest.event(profile, event);
             }
         } else if (execPeriod == null) {
+            register();
             new BukkitRunnable() {
                 @Override
                 public void run() {
+                    unregister();
```

Comment:
Registering and unregistering the `EventHandler` on the fly might cause bugs. (see previous comment)